rabbitmq-signadot-demo/publisher/app.py

```diff
@@ -0,0 +1,200 @@
+import json
+import logging
+import os
+import re
+from datetime import datetime
+
+import pika
+import redis
+from flask import Flask, request, jsonify
+
+app = Flask(__name__)
+logging.basicConfig(level=logging.INFO)
+
+# ---------------------------- Config & helpers ---------------------------- #
+
+def get_rabbitmq_port() -> int:
+    """
+    Accepts either a plain port (e.g. "5672") or a k8s-style "tcp://IP:PORT".
+    """
+    raw = os.environ.get("RABBITMQ_PORT", "5672")
+    if raw.startswith("tcp://"):
+        m = re.search(r":(\d+)$", raw)
+        if m:
+            return int(m.group(1))
+        logging.warning("Could not parse RABBITMQ_PORT '%s'; falling back to 5672", raw)
+        return 5672
+    try:
+        return int(raw)
+    except ValueError:
+        logging.warning("Could not parse RABBITMQ_PORT '%s'; falling back to 5672", raw)
+        return 5672
+
+RABBITMQ_HOST = os.environ.get("RABBITMQ_HOST", "localhost")
+RABBITMQ_PORT = get_rabbitmq_port()
+EXCHANGE_NAME = os.environ.get("CUSTOMER_EXCHANGE", "orders_exchange")
+AMQP_ROUTING_KEY = os.environ.get("AMQP_ROUTING_KEY", "orders")  # topic key used by exchange binding(s)
+
+logging.info("Using RabbitMQ at %s:%s", RABBITMQ_HOST, RABBITMQ_PORT)
+
+# Redis (optional)
+REDIS_HOST = os.environ.get("REDIS_HOST", "localhost")
+REDIS_PORT = int(os.environ.get("REDIS_PORT", "6379"))
+redis_client = None
+try:
+    redis_client = redis.Redis(host=REDIS_HOST, port=REDIS_PORT, decode_responses=True)
+    redis_client.ping()
+    logging.info("Connected to Redis at %s:%s", REDIS_HOST, REDIS_PORT)
+except Exception as e:
+    logging.warning("Redis not available: %s. Events will not be logged.", e)
+
+# When publisher runs inside a Signadot sandbox, this is injected.
+SANDBOX_ROUTING_KEY = os.environ.get("SIGNADOT_SANDBOX_ROUTING_KEY", "").strip()
+
+def log_event(event_type: str, data: dict) -> None:
+    if not redis_client:
+        return
+    try:
+        event = {"timestamp": datetime.utcnow().isoformat(), "type": event_type, "data": data}
+        redis_client.lpush("events", json.dumps(event))
+        redis_client.ltrim("events", 0, 999)  # keep last 1000
+    except Exception as e:
+        logging.error("Failed to log event to Redis: %s", e)
+
+def get_rabbitmq_channel():
+    conn = pika.BlockingConnection(pika.ConnectionParameters(host=RABBITMQ_HOST, port=RABBITMQ_PORT))
+    ch = conn.channel()
+    ch.exchange_declare(exchange=EXCHANGE_NAME, exchange_type="topic", durable=True)
+    return conn, ch
+
+def _extract_sd_routing_key(req) -> str:
```

Comment:
This propagation of baggage needs to be done using OpenTelemetry libraries. See example code here: https://docs.google.com/document/d/1qY0eutk-tFY-VkyMJC9ySrcFH5sLgr3tRVrOrNCL_g8/edit?tab=t.0#heading=h.sucvqambwijg

Similarly in the consumer code as well. This is automatically done by OTel so you don't need to do it manually.